test/e2e/network/firewall.go

```diff
@@ -218,18 +218,17 @@ var _ = common.SIGDescribe("Firewall rule", func() {
 
 		ginkgo.By("Checking well known ports on master and nodes are not exposed externally")
 		nodeAddr := e2enode.FirstAddress(nodes, v1.NodeExternalIP)
```

Comment:
what do we understand by externally here?
should they be reachable in the InternalIP?

Reply:
@aojea NodeExternalIP is usually supposed to be an IP that the node is reachable from, by the public internet or at least outside the cluster. Clearly nodes shouldn't be allowing anything in the world to talk to their kubelet port, so I think the test's intention is good. But it's definitely mishandling the case where the node doesn't have a NodeExternalIP, probably because Kube CI maybe always gives nodes one of those?

Reply:
the conclusion on the addresses semantic discussion is that the only certain address is the InternalIP https://github.com/kubernetes/kubernetes/pull/95768#issuecomment-822723560

I was wondering if the test should v1.NodeInternalIP, if those well known ports should not be exposed in that IP, of course

Reply:
not for this PR though